jax/_src/lax/control_flow.py

```diff
@@ -2574,39 +2574,27 @@ def _cumred_dtype_rule(name, operand, *args, **kw):
                     "of number.".format(name, np.dtype(operand.dtype).name))
   return dtypes.canonicalize_dtype(operand.dtype)
 
-cumsum_p = lax.standard_primitive(
-  _cumred_shape_rule, partial(_cumred_dtype_rule, "cumsum"),
-  'cumsum')
-ad.deflinear2(cumsum_p, _cumsum_transpose_rule)
-xla.backend_specific_translations['tpu'][cumsum_p] = xla.lower_fun(
-  partial(_cumred_tpu_translation_rule, lax._reduce_window_sum),
-  multiple_results=False)
-batching.primitive_batchers[cumsum_p] = partial(_cumred_batch_rule, cumsum_p)
-
-
-def _cumulative_reduction_primitive(name, reduce_window_fn):
+def _cumulative_reduction_primitive(name,
+                                    associative_reduce_window_fn,
```

Comment:
Just `reduce_fn`? There's no `window` about it if we are using `associative_scan`.

Reply:
Done